core/src/uix/compiler/aot.clj

```diff
@@ -446,6 +446,7 @@
 
 (defmethod compile-element :component [v]
   (let [[tag & args] v
+        tag (vary-meta tag assoc :tag 'js)
```

Comment:
Just curious, why can't we use regular type hints here?
```suggestion
  (let [[^js tag & args] v
```

Reply:
I don't know, but it doesn't work.

https://clojureverse.org/t/clojurescript-macros-and-type-hints/7292

Reply:
Ah, I see. Missed that we are inside a macro here. What is itching me a bit is the redefinition of `tag` inside the `let`…

```suggestion
  (let [[_tag & args] v
        tag (vary-meta (first v) assoc :tag 'js)
```

But nevermind, it doesn't get much better. Thank you for the explanation.

Reply:
I agree, usually, I don't do rebind the same symbol.